src/legacy/ui/public/visualize/components/visualization_chart.tsx

```diff
@@ -111,17 +102,15 @@ class VisualizationChart extends React.Component<VisualizationChartProps> {
       throw new Error('chartDiv and currentDiv reference should always be present.');
     }
 
-    const { vis, onInit } = this.props;
+    const { vis } = this.props;
     const Visualization = vis.type.visualization;
 
     this.visualization = new Visualization(this.chartDiv.current, vis);
 
-    if (onInit) {
-      // In case the visualization implementation has an isLoaded function, we
-      // call that and wait for the result to resolve (in case it was a promise).
-      const visLoaded =
-        this.visualization && this.visualization.isLoaded && this.visualization.isLoaded();
-      Promise.resolve(visLoaded).then(onInit);
+    // In case the visualization implementation has an isLoaded function, we
+    // call that and wait for the result to resolve (in case it was a promise).
+    if (this.visualization && this.visualization.isLoaded) {
+      this.visualization.isLoaded();
```

Comment:
The docs say we wait for the promise, but we don't wait for it here?

Reply:
i think we no longer need it, trying to remove (the only visualization which implements this is maps)